src/hashmap_v2.hpp

```diff
@@ -0,0 +1,177 @@
+        return *this;
+    }
+
+    void destroy() {
+        if (array_ != nullptr) {
+            for (int i = 0; i < array_size_; i++) {
+                LinkedList* current_node = array_[i].next_;
+                while (current_node != nullptr) {
+                    LinkedList* next_node = current_node->next_;
+                    delete current_node;
+                    current_node = next_node;
+                }
+            }
+            delete[] array_;
+        }
+    }
+
+    ~HashMapV2() { destroy(); }
+
+    void construct() { array_ = new LinkedList[array_size_]; }
+
+    // Random hash
+    size_t hash(const K& key) {
+        const unsigned char* data = reinterpret_cast<const unsigned char*>(&key);
+        size_t               size = sizeof(K);
+
+        size_t hash = 14695981039346656037ULL;
+        for (size_t i = 0; i < size; ++i) {
+            hash ^= static_cast<size_t>(data[i]);
+            hash *= 1099511628211ULL;
+        }
+
+        return hash;
+    }
+
+    void insert(const K& key, const V& value) {
+        if (array_ == nullptr) {
+            construct();
+        }
+        size_t      random_number = hash(key);
+        size_t      index         = random_number % array_size_;
+        LinkedList* current_node  = array_[index].next_;
+        while (current_node != nullptr) {
+            if (current_node->key_ == key) {
+                current_node->value_ = value;
+                return;
+            }
+            current_node = current_node->next_;
+        }
+        LinkedList* new_node = new LinkedList(key, value);
+        new_node->next_      = array_[index].next_;
+        array_[index].next_  = new_node;
+        elements_size_++;
+    }
+
+    bool find(const K& key, V& value) {
+        if (array_ == nullptr) {
+            return false;
+        }
+        size_t      random_number = hash(key);
+        size_t      index         = random_number % array_size_;
+        LinkedList* current_node  = array_[index].next_;
+        while (current_node != nullptr) {
+            if (current_node->key_ == key) {
+                value = current_node->value_;
+                return true;
+            }
+            current_node = current_node->next_;
+        }
+        return false;
+    }
+
+    bool remove(const K& key) {
+        if (array_ == nullptr) {
+            return false;
+        }
+        size_t      random_number = hash(key);
+        size_t      index         = random_number % array_size_;
+        LinkedList* current_node  = array_[index].next_;
+        LinkedList* previous_node = &array_[index];
+        while (current_node != nullptr) {
+            if (current_node->key_ == key) {
+                previous_node->next_ = current_node->next_;
+                delete current_node;
+                elements_size_--;
+                return true;
+            }
+            previous_node = current_node;
+            current_node  = current_node->next_;
+        }
```

Comment:
Let's remove this LinkedList node-delete logic out of here. Either use std::list, or create your own LinkedList with all the add/delete logic. Don't leak the linkedlist's delete logic into HashMap.